modules/gltf/gltf_document.cpp

```diff
@@ -6917,12 +6919,29 @@ Error GLTFDocument::append_from_file(String p_path, Ref<GLTFState> r_state, uint
 
 Error GLTFDocument::_parse_gltf_extensions(Ref<GLTFState> state) {
 	ERR_FAIL_NULL_V(state, ERR_PARSE_ERROR);
-	if (state->json.has("extensionsRequired") && state->json["extensionsRequired"].get_type() == Variant::ARRAY) {
-		Array extensions_required = state->json["extensionsRequired"];
-		if (extensions_required.find("KHR_draco_mesh_compression") != -1) {
-			ERR_PRINT("glTF2 extension KHR_draco_mesh_compression is not supported.");
-			return ERR_UNAVAILABLE;
+	if (state->json.has("extensionsUsed")) {
+		Array ext_array = state->json["extensionsUsed"];
+		state->extensions_used = ext_array;
+	}
+	if (state->json.has("extensionsRequired")) {
+		Array ext_array = state->json["extensionsRequired"];
+		state->extensions_required = ext_array;
+	}
+	Array supported_extensions;
+	supported_extensions.push_back("KHR_lights_punctual");
+	supported_extensions.push_back("KHR_materials_pbrSpecularGlossiness");
+	supported_extensions.push_back("KHR_texture_transform");
+	for (int32_t ext_i = 0; ext_i < document_extensions.size(); ext_i++) {
+		Ref<GLTFDocumentExtension> ext = document_extensions[ext_i];
+		ERR_CONTINUE(ext.is_null());
+		supported_extensions.append_array(ext->get_supported_extensions());
+	}
+	Error ret = Error::OK;
+	for (int i = 0; i < state->extensions_required.size(); i++) {
+		if (!supported_extensions.has(state->extensions_required[i])) {
+			ERR_PRINT("GLTF: Can't import file '" + state->filename + "', required extension '" + String(state->extensions_required[i]) + "' is not supported. Are you missing a GLTF extension?");
```

Comment:
"Are you missing a GLTF extension?" -> "Are you missing a GLTF extensions plugin?"

this could cause spam and lead to confusion on documents with lots of extensions (for example, .vrm files come bundled with as many as 5 extensions.)

possibly better: build a string with all missing extensions, comma separated, and construct the error message outside the for loop so it only gets printed once.

Reply:
Good suggestions, changed the text, but I did not change the printing to use a comma separated list. I really do not think this is a significant issue, simpler code is better here.